charts/consul/test/unit/gossip-encryption-autogeneration-job.bats

```diff
@@ -0,0 +1,74 @@
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      .
+}
+
+@test "gossipEncryptionAutogeneration/Job: enabled with global.gossipEncryption.autoGenerate=true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      --set 'global.gossipEncryption.autoGenerate=true' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "gossipEncryptionAutogeneration/Job: disabled when global.gossipEncryption.autoGenerate=false" {
+  cd `chart_dir`
+  assert_empty helm template \
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      --set 'global.gossipEncryption.autoGenerate=false' \
+      .
+}
+
+@test "gossipEncryptionAutogeneration/Job: fails if global.gossipEncryption.autoGenerate=true and global.gossipEncryption.secretName and global.gossipEncryption.secretKey are set" {
+  cd `chart_dir`
+  run helm template \
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      --set 'global.gossipEncryption.autoGenerate=true' \
+      --set 'global.gossipEncryption.secretName=name' \
+      --set 'global.gossipEncryption.secretKey=key' \
+      . 
+  [ "$status" -eq 1 ]
+  [[ "$output" =~ "If global.gossipEncryption.autoGenerate is true, global.gossipEncryption.secretName and global.gossipEncryption.secretKey must not be set." ]]
+}
+
+@test "gossipEncryptionAutogeneration/Job: fails if global.gossipEncryption.autoGenerate=true and global.gossipEncryption.secretName are set" {
+  cd `chart_dir`
+  run helm template \
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      --set 'global.gossipEncryption.autoGenerate=true' \
+      --set 'global.gossipEncryption.secretName=name' \
+      . 
+  [ "$status" -eq 1 ]
+  [[ "$output" =~ "If global.gossipEncryption.autoGenerate is true, global.gossipEncryption.secretName and global.gossipEncryption.secretKey must not be set." ]]
+}
+
+@test "gossipEncryptionAutogeneration/Job: fails if global.gossipEncryption.autoGenerate=true and global.gossipEncryption.secretKey are set" {
+  cd `chart_dir`
+  run helm template \
+      -s templates/gossip-encryption-autogen-job.yaml  \
+      --set 'global.gossipEncryption.autoGenerate=true' \
+      --set 'global.gossipEncryption.secretKey=key' \
+      . 
+  [ "$status" -eq 1 ]
+  [[ "$output" =~ "If global.gossipEncryption.autoGenerate is true, global.gossipEncryption.secretName and global.gossipEncryption.secretKey must not be set." ]]
+}
+
+
+@test "gossipEncryptionAutogeneration/Job: secretName and secretKey are generated" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/gossip-encryption-autogen-job.yaml \
+      --set 'global.gossipEncryption.autoGenerate=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.spec.containers[0].command | any(contains("secretName=RELEASE-NAME-consul-gossip-encryption-key\nsecretKey=key"))' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
```

Comment:
I don't think we can test this sufficiently in bats. Generally, if there's no templating logic in the command, then it won't be valuable to write a bats test for it. So I'd say remove this case altogether. 

It'd be nice to add this to at least our basic acceptance test. You'd just need to add it to helm values, setting
`global.gossipEncryption.autoGenerate` to `strconv.FormatBool(c.secure)` so that it's tested together with other secure features like TLS and ACLs:

https://github.com/hashicorp/consul-k8s/blob/9a974c1a9c9fe0594460352b25ef05fd4ba13734/charts/consul/test/acceptance/tests/basic/basic_test.go#L41-L45

Reply:
I started work on this, but ran into problems (I think I messed up my k8s cluster somehow). Can we sync on this and tackle it together next week?